lib/iopromise/data_loader.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module IOPromise
+  module DataLoader
+    module ClassMethods
+      def attr_promised_data(*args)
+        @promised_data ||= []
+        @promised_data.concat(args)
+
+        args.each do |arg|
+          self.class_eval("def #{arg};@#{arg}.sync;end")
+        end
+      end
+  
+      def promised_data_keys
+        @promised_data ||= []
```

Comment:
**Naming nitpick:** Renaming `@promised_data` to `@promised_data_keys` could be slightly less confusing.